deployment/helm/skaha/templates/session-volumes.yaml

```diff
@@ -0,0 +1,72 @@
+{{- define "skaha.session.commonVolumes" -}}
+      {{- if .Values.sessions.extraVolumes }}
```

Comment:
Personally I don't see the benefit of defining complex logic like this.
It is much simpler to just let the user define the desired volumes and volumeMounts with a direct definition in Helm values of arbitrary YAML, and just insert that YAML, like e.g. [this](https://gitlab.cern.ch/eos/eos-charts/-/merge_requests/50/diffs?commit_id=c863db520c1f35e4f7000af443f46a046ef171fb).  [Here](https://gitlab.cern.ch/eos/eos-charts/-/merge_requests/50/diffs?commit_id=a754e8e887e318bea4ef8ef8f3f5aa5f92f59dfd) is an example of what the usage looks like.

It is not necessary (or feasible) to enumerate every possible volume mounting option or type and have an if statement and corresponding value for each one; IMHO that adds unnecessary complexity to the chart. It is better to have complex data (structured YAML in your values file) than complex code.

That being said I'm not a science-platform maintainer, this is just a suggestion.

Reply:
If done in a fully general way like that it can be used to support any type of volume mount, so it should be possible to supersede/replace any specialized specific-purpose volume mounting config like in https://github.com/opencadc/science-platform/pull/645

Reply:
Thanks for the review, and I see your points @rptaylor   I agree with the aim of constraining the complexity.  Perhaps one benefit of this approach is that you don't have to re-apply your mounting config to the yaml launch files after a helm upgrade.

Reply:
Thanks for the review @rptaylor. We added this layer in order to avoid empty values and sanitise the input data. To avoid any failures regarding the same. User can just add their volumes via values.yaml, you can refer the example in sample-local-volumes.yaml.

Reply:
What's wrong with empty values?
Personally IMHO the k8s API is the best place to validate and sanitize input, rather than duplicating each field of the API in the Helm chart.  Further if statements would need to be added to support emptydirs, CSI volume types, various other options of configmaps and secrets, etc. (`kubectl explain pod.spec.volumes.secret; kubectl explain pod.spec.volumes.configMap`)

 It is also convenient to be able to just copy/paste the same format of YAML that is equally valid in a pod spec as in the helm values.  Anyway that is just my 2c.